### ISSUE TITLE
recovering from failed publish

### ISSUE THREAD

**Issue by pantoninho** [lerna/lerna#1668]
I'm having some troubles recovering from a failed publish.

After using `lerna version`, I called `lerna publish from-git`, which started publishing all my packages.
One of them failed (internet failure while uploading), and now I'm not able to resume the process.

If I call `lerna publish` it will fail as soon as it tries to re-publish one of the successfully published packages (due to publishing an already existing version).

In the past there was a `bail` option that allowed me to ignore failed commands and that was exactly what I was looking for, but looks like it was removed.

Is there a solution for this problem already?
Thank you

**Comment by evocateur** [comment]
Actually, `lerna publish` has never had a `--no-bail` option, unlike `lerna exec` or `lerna run`. Unfortunately, the "recovery" right now is, after it fails and leaves a mostly-dirty git tree, you have to go into the package directories one-by-one and run `npm publish <packageName>-<version>.tgz`. If the git tree has been cleaned already, you might be able to run `npm publish` in the leaf directories of the affected packages. And that ... really sucks.

#1648 gets closer to a "recovery" scenario that might help make this suck less. In an ideal world, recovering from a failed publish wouldn't require _ignoring_ failures, the command would be smart enough to skip packages that already have matching versions available in the configured registry.

**Comment by evocateur** [comment]
Yes, I would appreciate seeing what a solution could look like. I like the `from-package` positional argument I described in #1648 as a gate for the preflight version checks.
